server/ipqueue.go

```diff
@@ -195,6 +200,9 @@ func (q *ipQueue[T]) popOne() (T, bool) {
 		default:
 		}
 	} else {
+		if q.calc != nil {
+			q.sz -= q.calc(e)
```

Comment:
This is not needed as `q.sz` is about to be set to zero anyway.

Reply:
So basically, these changes can be reduced to the use of `defer`. But although I think that `defer` have become very efficient, I think it should be left the way it was (to make it even more efficient ;-)).